backend/python/app/services/implementations/story_service.py

```diff
@@ -595,6 +595,49 @@ def approve_all_story_translation_content(self, story_translation_id):
             )
             raise error
 
+    def finish_grading_story_translation(
+        self, test_result, test_feedback, story_translation_test_id
+    ):
+        try:
+            if "translate" not in test_result or "review" not in test_result:
+                raise Exception(
+                    "Error: test_feedback must contain translate and review attributes."
+                )
+
+            story_translation_test = (
+                db.session.query(StoryTranslation)
+                .filter(StoryTranslation.id == story_translation_test_id)
+                .first()
+            )
+
+            story_translation_contents = (
+                db.session.query(StoryTranslationContent)
+                .filter(
+                    StoryTranslationContent.story_translation_id
+                    == story_translation_test_id
+                )
+                .all()
+            )
+
+            story_translation_test.test_feedback = test_feedback
+            story_translation_test.test_result = test_result
+            story_translation_test.test_grade = self._get_final_grade(
+                story_translation_contents
+            )
+            story_translation_test.stage = "PUBLISH"
+            db.session.commit()
```

Comment:
nit: can you leave a TODO to set the story_translation_test.reviewer_id to the admin's id?

Reason I'm asking for a TODO comment rather than actually doing it in this pr: ever since we struggled with the unit tests for the comments, I've been thinking we need to rethink how we authenticate and pass in user_ids to services.

Reply:
oh shit wait on second thought, take a look at how get_user_id_from_request is used in user_mutation's UpdateMe
Can you get the user's id and set the reviewer_id here?